pyvista/plotting/mapper.py

```diff
@@ -854,6 +854,60 @@ def scale_factor(self) -> float:
     def scale_factor(self, value: float):
         return self.SetScaleFactor(value)
 
+    @property
+    def scale_array(self) -> str:
+        """Set or return the name of the array used to scale the splats.
+
+        Scalars used to scale the gaussian points. Accepts a string
+        name of an array that is present on the mesh.
+
+        Parameters
+        ----------
+        name : str
+            Name of the array used to scale the points.
+
+        Notes
+        -----
+        Setting this automatically sets ``scale_factor = 1.0``.
+
+        Examples
+        --------
+        Plot spheres using `style='points_gaussian'` style and scale them by
+        radius.
+
+        >>> import numpy as np
+        >>> import pyvista as pv
+        >>> n_spheres = 1_000
+        >>> pos = np.random.random((n_spheres, 3))
+        >>> rad = np.random.random(n_spheres) * 0.01
+        >>> pdata = pv.PolyData(pos)
+        >>> pdata['radius'] = rad
+        >>> pl = pv.Plotter()
+        >>> actor = pl.add_mesh(
+        ...     pdata,
+        ...     style='points_gaussian',
+        ...     emissive=False,
+        ...     render_points_as_spheres=True,
+        ... )
+        >>> actor.mapper.scale_array = 'radius'
+        >>> pl.show()
+
+        """
+        self.scale_factor = 1.0
```

Comment:
VTK specifically allows for both of these settings to be applied independently, and the default for scale_factor is already 1.0 in vtk, but I haven't chased down the default in the pyvista code.  This is a very useful feature.  For example, I have data from a simulation that has particle size in mm stored in point data as "diameter", but the mesh positions are in units of m.  I would want to set the scale_factor to 0.001 and scale_array to "diameter" to render this correctly.

Reply:
Ah yes, this shouldn't be hard set here

Reply:
And this is in the getter? Seems like a mistake

Reply:
```suggestion
```

Reply:
The function `use_circular_spat` does `self.scale_factor *= 1.5` so I wanted to cancel it at first without modifying the function, I think it must be done somewhere

Reply:
This is a bit of an interconnected rabbit hole.

This line seems mysterious to me:
https://github.com/pyvista/pyvista/blob/d9dd94e8999c6eac0a491af50439574ecb4368f3/pyvista/plotting/plotting.py#L3505

Then there is this line, which as @kmarchais points out further modifies `scale_factor` if a different option is used:
https://github.com/pyvista/pyvista/blob/d9dd94e8999c6eac0a491af50439574ecb4368f3/pyvista/plotting/plotting.py#L3512